src/lib.rs

```diff
@@ -376,12 +395,24 @@ impl CommitLog {
     }
 
     /// Appends log entrites to the commit log, returning the offsets appended.
+    ///
+    /// If the message size exceeds the limit set in the options an error is
+    /// returned.
+    ///
+    /// ```
+    /// # use commitlog::*;
+    /// let mut opts = LogOptions::new("log");
+    /// opts.message_max_bytes(100);
+    /// let mut log = CommitLog::new(opts).unwrap();
+    /// assert!(log.append_msg([1u8; 100].as_ref()).is_ok());
+    /// assert!(log.append_msg([1u8; 101].as_ref()).is_err());
+    /// ```
     pub fn append<T>(&mut self, buf: &mut T) -> Result<OffsetRange, AppendError>
     where
         T: MessageSetMut,
     {
-        //Check if given message exceeded the max size
-        if buf.bytes().len() > self.file_set.log_options().message_max_bytes {
+        // Check if given message exceeded the max size
+        if buf.bytes().len() > (self.file_set.log_options().message_max_bytes + MSG_HEADER_LEN) {
```

Comment:
We'd need to multiply `MSG_HEADER_LEN` by the number of items in the message set. At least currently, each message gets a header.